lib/src/data/datasource/remote/pokedex_api_service.dart

```diff
@@ -50,10 +55,47 @@ class PokedexApiService {
       if (response.statusCode == HttpStatus.ok) {
         return PokemonModel.fromJson(json.decode(response.body));
       } else {
-        return PokemonModel(sprites: SpritesModel());
+        return PokemonModel(
+          sprites: SpritesModel(),
+          species: SpeciesModel(
+            color: ColorModel(),
+            habitat: HabitatModel(),
+            shape: ShapeModel(),
+          ),
+        );
+      }
+    } catch (exception) {
+      return PokemonModel(
+        sprites: SpritesModel(),
+        species: SpeciesModel(
+          color: ColorModel(),
+          habitat: HabitatModel(),
+          shape: ShapeModel(),
+        ),
+      );
+    }
```

Comment:
Try to omit the creation of default models when the service fail, you can change this "Future<PokemonModel>" to "Future<PokemonModel?>"

Reply:
got it